### PR TITLE
Again, more QtQuick 1 extermination: revamped variable editor

### PR DESCRIPTION
**Pull request by nirvn**
The last use of QtQuick 1's TreeView is gone:
Screenshot from 2020-04-28 11-14-15

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-d8abf91da882cb81e10e07fc57b472ff90b1e557-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-d8abf91da882cb81e10e07fc57b472ff90b1e557-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-d8abf91da882cb81e10e07fc57b472ff90b1e557-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-d8abf91da882cb81e10e07fc57b472ff90b1e557-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-9b003f6b44caa0e6b1012c0958161dfcd6b1f37e-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-9b003f6b44caa0e6b1012c0958161dfcd6b1f37e-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-9b003f6b44caa0e6b1012c0958161dfcd6b1f37e-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-9b003f6b44caa0e6b1012c0958161dfcd6b1f37e-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-bac65b48ccc2d54581fbd0f89f713bf2c71b2017-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-bac65b48ccc2d54581fbd0f89f713bf2c71b2017-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-bac65b48ccc2d54581fbd0f89f713bf2c71b2017-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-bac65b48ccc2d54581fbd0f89f713bf2c71b2017-x86_64.apk)

File: src/core/expressionvariablemodel.cpp [hunk 1]
```diff
@@ -79,8 +79,12 @@ void ExpressionVariableModel::reloadVariables()
     if ( scope->isReadOnly( varName ) )
     {
       QStandardItem *nameItem = new QStandardItem( varName );
+      QVariant varValue = scope->variable( varName );
+      if ( varValue == QStringLiteral( "Not availble" ) )
```
Comment (m-kuhn) [inline src/core/expressionvariablemodel.cpp:83]:
```suggestion
      if ( varValue == QStringLiteral( "Not Available" ) )
```
?

File: src/core/expressionvariablemodel.cpp [hunk 2]
```diff
@@ -79,8 +79,12 @@ void ExpressionVariableModel::reloadVariables()
...
+        varValue = QVariant( QT_TR_NOOP( "Not available" ) );
```
Comment (m-kuhn) [inline src/core/expressionvariablemodel.cpp:84]:
```suggestion
        varValue = QVariant( QT_TR_NOOP( "Not Available" ) );
```

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-5e19f4179b68a3c4aa88652ae833d7d530b276d5-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-5e19f4179b68a3c4aa88652ae833d7d530b276d5-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-5e19f4179b68a3c4aa88652ae833d7d530b276d5-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-5e19f4179b68a3c4aa88652ae833d7d530b276d5-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-18090991fc8eb8bce2dcbe8ed6bb9096cb0e6fcb-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-18090991fc8eb8bce2dcbe8ed6bb9096cb0e6fcb-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-18090991fc8eb8bce2dcbe8ed6bb9096cb0e6fcb-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-971-18090991fc8eb8bce2dcbe8ed6bb9096cb0e6fcb-x86_64.apk)

**Comment by nirvn** [comment]
@m-kuhn , review addressed, merge?

**Pull request merged**